In [24]:
from collections import Counter
import collections
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [3]:
data = pd.read_csv('footdata.csv')

In [4]:
data.head()

Date  HomeTeam    AwayTeam  FTHG  FTAG FTR  HTHG  HTAG HTR  HS  ...  \
0  04/08/07  Bordeaux        Lens     1     0   H     1     0   H  13  ...   
1  04/08/07      Caen        Nice     1     0   H     0     0   D  16  ...   
2  04/08/07     Lille     Lorient     0     0   D     0     0   D   6  ...   
3  04/08/07    Monaco  St Etienne     1     1   D     1     0   H  16  ...   
4  04/08/07  Paris SG     Sochaux     0     0   D     0     0   D  11  ...   

   AF  HC  AC  HY  AY  HR  AR  B365H  B365D  B365A  
0  19   4   4   3   2   0   1   2.20   2.87    3.6  
1  15   2   3   1   1   0   0   2.40   2.75    3.0  
2  19   2   2   1   1   0   0   1.72   3.00    5.0  
3   3   5   6   3   3   0   0   1.90   2.90    4.0  
4  17   5   9   2   1   0   0   1.72   3.40    5.0  

[5 rows x 24 columns]

Ce jeu de données contient l’ensemble des matches du championnat de France de football 2007-2008. Composé de 20 équipes s’affrontant chacune lors de rencontres allers et retours, le tableau de données contient les résultats de 380 matches. Le tableau est composé des colonnes suivantes :

+ Date : Date du match
+ AST : Tirs cadrés E2
+ HomeTeam : Equipe à domicile (E1)
+ HF : Fautes commises par E1
+ AwayTeam : Equipe à l’extérieur (E2)
+ AF : Fautes commises par E2
+ FTHG : Buts E1
+ HC : Corners obtenus par E1
+ FTAG : Buts E2
+ AC : Corners obtenus par E2
+ FTR : Résultat
+ HY : Cartons jaunes E1
+ HTHG : Buts E1 à la mi-temps
+ AY : Cartons jaunes E2
+ HTAG : Buts E2 à la mi-temps
+ HR : Cartons rouges E1
+ HTR : Résultat à la mi-temps
+ AR : Cartons rouges E2
+ HS : Tirs E1
+ B365H : Côte sportive victoire E1
+ AS : Tirs E2
+ B365D : Côte sportive match nul
+ HST : Tirs cadrés E1
+ B365A : Côte sportive victoire E2

In [5]:
win_counter = Counter(data['FTR'])
win_counter

Counter({'H': 167, 'D': 116, 'A': 97})

Il y'a eu 167 victoires à domicile, 116 matchs nuls et 97 victoire en exterieur

In [6]:
hometeams_win = data['HomeTeam'][data['FTR']=='H'] #toutes les fois ou l'équipe a domicile gagne
awayteams_win = data['AwayTeam'][data['FTR']=='A'] #toutes les fois ou l'équipe a l'exterieur gagne

In [7]:
home_win_counter = Counter(hometeams_win)
away_win_counter = Counter(awayteams_win)
total_win_counter = home_win_counter + away_win_counter

In [26]:
sorted_dict = collections.OrderedDict((sorted(total_win_counter.items(), key=lambda t: t[1])))

In [27]:
sorted_dict

OrderedDict([('Metz', 5),
             ('Toulouse', 9),
             ('Strasbourg', 9),
             ('Lens', 9),
             ('Sochaux', 10),
             ('Paris SG', 10),
             ('Valenciennes', 12),
             ('Lorient', 12),
             ('Auxerre', 12),
             ('Caen', 13),
             ('Nice', 13),
             ('Monaco', 13),
             ('Lille', 13),
             ('Le Mans', 14),
             ('Nancy', 15),
             ('St Etienne', 16),
             ('Rennes', 16),
             ('Marseille', 17),
             ('Bordeaux', 22),
             ('Lyon', 24)])

C'est Lyon qui totalise le plus de victoires avec 24 victoires, metz a le moins de victoires avec 5 victoires seulement

In [ ]:
plt.figure(figsize=(10, 5))
ax = sns.countplot(hometeams_win, order=hometeams_win.value_counts().index)
plt.setp(ax.get_xticklabels(), rotation=90)[1]
plt.show()

C'est Lyon qui a le plus de victoires à domicile

In [ ]:
plt.figure(figsize=(10, 5))
ax = sns.countplot(awayteams_win, order=awayteams_win.value_counts().index)
plt.setp(ax.get_xticklabels(), rotation=90)[1]
plt.show()

C'est également Lyon qui a le plus de victoires en extérieur

In [ ]:
hometeams_loss = data['HomeTeam'][data['FTR']=='A'] #toutes les fois ou l'équipe a domicile perd
awayteams_loss = data['AwayTeam'][data['FTR']=='H'] #toutes les fois ou l'équipe a l'exterieur perd

In [ ]:
home_loss_counter = Counter(hometeams_loss)
away_loss_counter = Counter(awayteams_loss)
total_loss_counter = home_loss_counter + away_loss_counter

In [ ]:
total_loss_counter

l'équipe qui perd le plus est Metz avec 24 défaites

In [ ]:
#defaites a domicile
plt.figure(figsize=(10, 5))
ax = sns.countplot(hometeams_loss, order=hometeams_loss.value_counts().index)
plt.setp(ax.get_xticklabels(), rotation=90)[1]
plt.show()

L'équipe qui a le plus de défaites à domicile est Metz

In [ ]:
#defaites en extérieur
plt.figure(figsize=(10, 5))
ax = sns.countplot(awayteams_loss, order=awayteams_loss.value_counts().index)
plt.setp(ax.get_xticklabels(), rotation=90)[1]
plt.show()

L'équipe qui a le plus de défaites en extérieur est Lens (cette équipe a cependant peu de défaites à domicile)

In [ ]:
home_shots = data[['HomeTeam', 'FTHG', 'HS', 'HST']] #buts, tirs, tirs cadrés E1
away_shots = data[['AwayTeam', 'HTAG', 'AS', 'AST']] #buts, tirs, tirs cadrés E2
home_shots.columns = ['team', 'buts', 'tirs', 'tirs cadrés']
away_shots.columns = ['team', 'buts', 'tirs', 'tirs cadrés']

In [ ]:
total_shots = pd.concat([home_shots, away_shots]) #on combine les deux dataframes pour avoir les stats totals domicile et ext
total_shots = total_shots.groupby('team').sum() #nb de buts, tirs et tirs cadrés pour chaque équipe

total_shots

l'équipe qui a le plus de buts est Lyon, l'équipe qui a le plus de tirs est St Etienne et l'équipe qui a le plus de tirs cadrés est Lyon